roles/ara_api/tasks/install/pypi.yaml

```diff
@@ -0,0 +1,33 @@
+---
+- name: Install required packages from Pypi
+  pip:
+    name: "{{ pip_pkg }}"
+    virtualenv: "{{ ara_api_venv_path }}"
+    state: present
+    virtualenv_command: /usr/bin/python3 -m venv
+  loop:
+    - django
+    - django-cors-headers
+    - django-filter
+    - django-health-check
+    - django-utils
+    - djangorestframework
+    - dynaconf
+    - health-check
+    - pygments
+    - tzlocal
+    - whitenoise
+  loop_control:
+    loop_var: pip_pkg
+
+- name: Prefix the virtualenv bin directory to PATH
+  set_fact:
+    path_with_virtualenv: "{{ ara_api_venv_path }}/bin:{{ ansible_env.PATH }}"
+  when: ara_api_venv | bool
+
+- name: Install ARA from Pypi
+  pip:
+    name: ara
+    version: "{{ ara_api_version }}"
+    virtualenv: "{{ ara_api_venv_path }}"
+    state: present
```

Comment:
hello o/

I understand the intent of the playbook to be to install ara and it's API server dependencies from pypi in a virtualenv.

It turns out that we could use something like this in the ara_api role -- right now it does this but from a git checkout instead: https://github.com/ansible-community/ara-collection/blob/36ac8f835c37c83f402161c77ad63395a8f03a5f/roles/ara_api/tasks/install/source.yaml#L25-L30

I think we could have a ``roles/ara_api/install/pypi.yaml`` with that approach and then it'd be automatically available when selecting ``ara_api_install_method: pypi``:
https://github.com/ansible-community/ara-collection/blob/36ac8f835c37c83f402161c77ad63395a8f03a5f/roles/ara_api/defaults/main.yaml#L42-L46

Reply:
That's the purpose of this playbook :-) Below is a snippit from the actual code I use atm:

```
ara_roles__to_merge:
  - 'recordsansible.ara.ara_api'
  - 'recordsansible.ara.ara_web'

ara_api_install_method: 'pypi'
ara_api_version: 'latest'
```

Combined with my version of the collection, this works like a charm!

And I have a co-worker of mine working on the same for ara_web's NPM dependencies

Reply:
:+1:

ara-web is unfortunately neglected and behind on features nowadays, it could definitely use some love, thanks :)